### PR TITLE
Frontend/licensee dashboard

### LINKED ISSUE #202: Applicant dashboard
- reference applicant dash in Figma
- no need to include adverse action info or obtain privilege
- getMe endpoint
- when a licensee logs in, they see the dashboard
- return expired licenses and privileges

### PR DESCRIPTION
**Pull request by ChiefStief**
### Requirements List
- Make sure you have a licensee user in your sandbox with data populated such that something will be returned on the `GET /provider-users/me` end point. Get help from Landon or Justin if required.

### Description List
- Implemented licensee dashboard screen and associated components shown here: https://www.figma.com/design/SYM0uWszsu8Sf0YfxAhIMY/JCC?node-id=6812-1373&node-type=section&t=dxbOfq8uEOCClflS-0
- Edited login flow for licensees to make functional with addition of compact checking
- Refactored user model into 3 Classes + added new functionality, User (common functionality), StaffUser (extends user), and LicenseeUser (extends user)
- Made edits associated with User model refactor
- Added get LicenseeUser network call and altered authToken functionality to set the correct auth header depending on which endpoint we are hitting
- Updated Tests

### Testing List
- `yarn test:unit:all` should run without errors or warnings
- `yarn serve` should run without errors or warnings
- `yarn build` should run without errors or warnings
- Code review
- Smoke test logging in as a provider with the mockAPI off and viewing licensee dashboard screen. Confirm flow works and looks as expected
- Smoke test logging in as a provider with the mockAPI off and viewing licensee dashboard screen. Confirm flow works and all edge cases / possible UI variations are represented and looks good

Closes #202

### CODE REVIEW

**Comment by jlkravitz** [comment]
- [ ] review the pull request to get oriented
	- [ ] read the description of the pull request, which should summarize the changes made
	- [ ] read through every task on the Scrum board that's encompassed by this pull request
	- [ ] read the description of the commits that comprise the pull request
- [ ] skim all new code, in the context of existing code, looking for problems (knowing that the vast majority of new code will be covered by tests)
- [ ] review all tests
	- [ ] methodically review all new tests for correctness, quality of naming
	- [ ] look at code coverage of tests
	- [ ] determine what code isn’t tested, review that rigorously
- [ ] review documentation to ensure that it matches changes
- [ ] provide comments on the pull request on GitHub, as necessary
	- [ ] for comments that are specific to a particular line of code, comment on those specific lines
	- [ ] for comments that are more general, attach the comment to a random line in `README.md` (as opposed to commenting on the pull request itself), to be able to use GitHub's ability to thread discussions on those comments

File: webroot/src/components/PrivilegeCard/PrivilegeCard.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+@Component({
+    name: 'PrivilegeCard',
+})
+class PrivilegeCard extends Vue {
+    @Prop({ required: true }) privilege?: License;
+
+    //
+    // Computed
+    //
+    get statusDisplay():string {
```
Comment (jlkravitz) [inline webroot/src/components/PrivilegeCard/PrivilegeCard.ts:26]: same here on spacing? 
```suggestion
    get statusDisplay(): string {
```

File: webroot/src/models/Licensee/Licensee.model.ts [hunk 2]
```diff
@@ -230,8 +230,14 @@ export class LicenseeSerializer {
...
         // In get-all responses, server only returns a license state, not the actual license objects
-        if (json.licenseJurisdiction) {
+        if (json.licenseJurisdiction && json.licenses && json.licenses.length < 2) {
             licenseeData.licenseStates.push(new State({ abbrev: json.licenseJurisdiction }));
+        } else if (json.licenses && json.licenses.length > 1) {
+            json.licenses.forEach((serverLicense) => {
+                if (serverLicense.status === 'active') {
+                    licenseeData.licenseStates.push(new State({ abbrev: serverLicense.jurisdiction }));
+                }
+            });
```
Comment (jlkravitz) [inline webroot/src/models/Licensee/Licensee.model.ts:240]: More to test my understanding, but couldn't we just run this code in all cases? I assume now that we're returning all the licenses, `licenseJurisdiction` is sort of duplicative?

File: webroot/src/models/Licensee/Licensee.model.ts [hunk 3]
```diff
@@ -230,8 +230,14 @@ export class LicenseeSerializer {
         };
 
         // In get-all responses, server only returns a license state, not the actual license objects
```
Comment (jlkravitz) [inline webroot/src/models/Licensee/Licensee.model.ts:232]: Is this comment still true?

File: webroot/src/network/userApi/interceptors.ts [hunk 4]
```diff
@@ -14,9 +14,18 @@ import { authStorage, tokens } from '@/app.config';
...
+    let authToken;
+    let authTokenType;
+
+    if (licenseeUserEndPoints.includes(url)) {
+        authToken = authStorage.getItem(tokens.licensee.ID_TOKEN);
+        authTokenType = authStorage.getItem(tokens.licensee.AUTH_TOKEN_TYPE);
+    } else {
+        authToken = authStorage.getItem(tokens.staff.AUTH_TOKEN);
+        authTokenType = authStorage.getItem(tokens.staff.AUTH_TOKEN_TYPE);
+    }
```
Comment (jlkravitz) [inline webroot/src/network/userApi/interceptors.ts:28]: Can you add a comment briefly explaining this? I _think_ the intent is to handle cases when someone is both a staff and a licensee?

Also, is there a conceptual reason it's called `ID_TOKEN` for the licensee and `AUTH_TOKEN` for the staff?

File: webroot/src/pages/LicenseeDashboard/LicenseeDashboard.ts [hunk 5]
```diff
@@ -6,10 +6,112 @@
...
+
+    get licenseExpiredMessage(): string {
+        return this.$t('licensing.licenseExpiredMessage');
+    }
+
+    //
+    // Methods
+    //
+    startPrivPurchaseFlow() {
+        console.log('Starting Privilege Purchase Flow!');
```
Comment (jlkravitz) [inline webroot/src/pages/LicenseeDashboard/LicenseeDashboard.ts:105]: Is this unimplemented?

**Review by jlkravitz — COMMENTED** [review]
a few comments but generally looks great!

File: webroot/src/pages/LicenseeDashboard/LicenseeDashboard.ts (hunk 5 above)
Comment (ChiefStief) [reply webroot/src/pages/LicenseeDashboard/LicenseeDashboard.ts:105]: Yes, next sprint we will be implementing the privilege purchase flow. This seemed like the correct cut off point to not start that feature but complete this one.

File: webroot/src/components/LicenseCard/LicenseCard.ts [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+        }
+
+        return licenseStatus;
+    }
+
+    get isActive(): boolean {
+        return Boolean(this.license && this.license.statusState === LicenseStatus.ACTIVE);
+    }
+
+    get stateTitle():string {
```
[the remark this answers is not in the document]
Comment (ChiefStief) [reply webroot/src/components/LicenseCard/LicenseCard.ts:40]: Yea, it seems our linter isn't strict about this particular thing. Ill mention it to John and see if we can add this.

File: webroot/src/network/userApi/interceptors.ts (hunk 4 above)
Comment (ChiefStief) [reply webroot/src/network/userApi/interceptors.ts:28]: I will.

Essentially, we have two different user pools that produce different unrelated auth tokens depending on if the user is a licensee or a staff user. Storing the tokens in different places and treating them as ignorant of each other is a knock-on effect of this, the "two app" approach and the edge cases that involve a user going through the sign in flow for both user types.

Certain endpoints are designed to be called by staff users and certain endpoints are designed to be called by licensee users. We need to send the correct token (determined by which endpoint we are hitting) up with the call.

I dont know the underlying reason why the licensee endpoints want the Id token and the staff endpoints want the auth token. Landon or Justin would be better able to answer that question.

File: webroot/src/models/Licensee/Licensee.model.ts (hunk 3 above)
Comment (ChiefStief) [reply webroot/src/models/Licensee/Licensee.model.ts:232]: Confirming with the team but I think there actually weren't changes here. However, Im glad you pointed this out because I should not have touched this file and must have forgotten to clean up after I was prototyping certain things. As a result I have moved a portion of the functionality to the UI presentation layer where it should have been from the beginning.

Comment (ChiefStief) [reply webroot/src/models/Licensee/Licensee.model.ts]: Yea both cases still need to be there

**Review by jlkravitz — APPROVED** [review]
@isabeleliassen good to go!

**Review by isabeleliassen — APPROVED** [review]
